src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -113,12 +113,7 @@ def _smooth_prediction(self, pred_rv, loc):
         next_idx = (self.locations < loc).sum()
         next_loc = self.locations[next_idx]
         next_rv = self._state_rvs[next_idx]
-        next_pred, crosscov = self.gauss_filter.predict(
-            start=loc, stop=next_loc, randvar=pred_rv
-        )
-        smoothed_rv = self.gauss_filter.smooth_step(
-            pred_rv, next_pred, next_rv, crosscov
-        )
+        smoothed_rv = self.gauss_filter.smooth_step(pred_rv, next_rv, loc, next_loc)
```

Comment:
this code was incorrect before this change. We tested it, but not the values, just the shape... We still do, because I am not sure how to test this well, but at least it is correct now :)

Reply:
I'm very surprised that tests passed, if the previous version was incorrect. That didn't even really match the function signature??

Reply:
Me too. Was this even tested properly?

Reply:
Well it seems it's even not covered right now so I guess it was not :see_no_evil:

Reply:
lets open an issue about this (and the lack of tests below, too). Seems like we have to extend the test coverage of kalmanposterior a bit.